### ISSUE TITLE
Simplified installation

### ISSUE THREAD

**Issue by vlcinsky** [asoroa/ical2org.py#20]
Current installation has issues such as need to install it into global python interpreter or in case of installation into virtualenv it has broken shebang pointing to global python version.

The solution may be to:
- make the script proper python package
- name the package e.g. ical2orgpy (as ical2org is already taken by another tool, working just on Macs)
- the package would create command ical2orgpy

The package would:
- include `tox.ini` for testing (with python 2.7, 3.4+)
- setup.py
- other source code

I am installing the packages this way (on Linux):
Assume python and tox are installed.
```bash
$ sudo su
# cd /opt
# mkdir ical2orgpy
# cd ical2orgpy
# tox-quickstart
....
```
Finally the `tox.ini` looks like:
```ini
[tox]
envlist = py35 # use the version which is supported on both system and by the package
skipsdist = True

[testenv]
deps = ical2orgpy
```
Then continue with installing the script into virtualenv:
```
# tox
```
The script shall be now at `/opt/ical2orgpy/.tox/py35/bin/ical2orgpy` and it shall have all required dependencies available there.

Final step is to make it globally visible, e.g. by symlinking to `/usr/local/bin`:
```
# source .tox/py35/bin/activate
(py35)# cd /usr/bin/local
(py35)# ln -s `which ical2orgpy`
(py35)# deactivate
# exit
```
From now on, the command `ical2orgpy` shall be available anywhere in the system.

If you feel interested, I may prepare such pull request (but bear with me, it may take a while till I find the time to complete it).

Next actions would be:
- you agree on package name (my proposal is `ical2orgpy`)
- I create pull request with python package (`setup.py` and all that stuff around)
- after you accept the PR, you would register the package at pypi.python.org and push the package there (I would provide instructions).

**Comment by asoroa** [comment]
Thanks for the interest, I'm definitively interested. Please go ahead and prepare the PR.

**Comment by vlcinsky** [comment]
See PR #21

**Comment by vlcinsky** [comment]
@asoroa your question "how to install the python package based script easily" lead me to discovery of nice tool called `pipsi`. Here are the instructions for installation and few words about why python package and why to bother with virtualenvs.

# TL;DR

Installing CLI provided via python package may be as easy as:

```
$ curl https://raw.githubusercontent.com/mitsuhiko/pipsi/master/get-pipsi.py | python
$ cd {ical2orgpy-project-directory}
$ pipsi install .
$ ical2orgpy --help
Usage: ical2orgpy [OPTIONS] ICS_FILE ORG_FILE

  Convert ICAL format into org-mode.

  Files can be set as explicit file name, or `-` for stdin or stdout::

      $ ical2orgpy in.ical out.org

      $ ical2orgpy in.ical - > out.org

      $ cat in.ical | ical2orgpy - out.org

      $ cat in.ical | ical2orgpy - - > out.org

Options:
  -p, --print-timezones     Print acceptable timezone names and exit.
  -d, --days INTEGER RANGE  Window length in days (left & right from current
                            time). Has to be positive.
  -t, --timezone TEXT       Timezone to use. (local timezone by default)
  -h, --help                Show this message and exit.
```

This assumes:
- you have some python installed
- directory `~/.local/bin` is searchable thus included in $PATH

The installation is "clean one", what means, that the package is installed into separate virtual environment (located at `~/.local/venv/ical2orgpy`), thus not spoiling global or user python environment. To make the CLI easy to access, it is symlinked into `~/.local/bin` directory.

The same shall (with small modifications) work for MS Windows and for MacOS.

`pipsi` allows installig any number of python packages, listing them, perform upgrades as well as uninstallation and each package can use indepent version of python if that is required.

# Why to bother with Python virtual environments
Scripts can be installed into different python environments:

- global Python environment
- user Python environment
- application Python environment

While installing into *global environment* sounds easy, it can quickly become a nightmare as one has to install other python packages, required by the tool what is sometime easy as `$ sudo apt install python-docutils`, but sometime proper package is missing or is the application requires newer version. Using `$ sudo pip install {whatever-package}` may work, but may also break whole system or at least a part of it.

With *user environment* the risk of breaking the whole system is minimal, but one can soon find, that some of packages to install are conflicting.

With *application environment* one has all the things under control as each application has it's own private virtual environment not conflicting with others.

# Why not plain Python based script

Some things are easily done by single file run by `$ python my-script.py` or using sheban `#!env python`. This works easily until one needs to install a package required by the script. At that moment we have the same problem as with installing into global or user python environment.

Other issues are:
- install/uninstall/upgrade
- testing

# How is python based CLI installed
It is easy to understand, how plain python script looks and how it is run.

With Python package based it a bit different, but can be understood too.

On Linux, the python package installation puts the CLI in form of python script with shebang into `bin` directory of current python environment. As the shebang points to the python interpreter by absolut path, it is easy to move or symlink the file to a location, wich is searchable (included in PATH).

On MS Windows (as far as I remember) the CLI is installed in form of two files. One is executable and the other one is actual python code, both files having the same basename (they differ only in the suffix). The executable is started, and tries to start the python file having the same name as the executable itself only with an extension `.py`. One can do some experiments with renameing both files and see, all works well (all the executables for different CLI have the same size).

# No need to activate virtual envs all the time
The activation of virtualenv is needed only during interactive development. In fact, the environment is determined by the python interpreter in charge, which can be done explicitly by calling given python, or by calling a script, pointing to given python by shebang.

# Typical procedure to install into virtual environment

The steps typicaly are:
- create virtualenv (using `venv`, `virtualenvwrapper`, `tox`...)
- activate it (mostly `$ source venv/bin/activate`)
- install the tool (`$ pip install {the-tool}`)

To make the CLI, it is handy to create symlinks from the actual script (typicali present at the same directory the `activate` script is present) into a directory, which is included in `PATH`.

# Tools to make installation easy

- `pipsi`: shown above (the simplest method for deployment)
- `virtualenvwrapper`: convenient for deployment as well as for development
- `tox`: great for repeatable deployment based on few files (`tox.ini` and `requirements.txt`)

Details about these tools is out of scope.

**Comment by vlcinsky** [comment]
@asoroa I still do not see the package on pypi.python.org so I assume you are going step by step to completing the release.

In case the publishing to pypi seems a problem, I can offer either detailed instructions (as published another package [pytest-xfiles](https://pypi.python.org/pypi/pytest-xfiles/0.1.1) a week ago) or manage publishing the package myself.

**Comment by asoroa** [comment]
@vlcinsky I just uploaded the package [here](https://pypi.python.org/pypi/ical2orgpy/0.2.0). Thanks to your great job it was quite easy to do :-)

**Comment by vlcinsky** [comment]
@asoroa Great.

Now it goes as easy as (assuming [pipsi](https://github.com/mitsuhiko/pipsi) is installed):

```bash
$ pipsi install ical2orgpy
...
$ which ical2orgpy
~/.local/bin/ical2orgpy
```

and it is installed in virtualenv.

Well done. Closing.
